apps/emqx/src/emqx_persistent_session_ds_state.erl

```diff
@@ -359,17 +369,18 @@ del_rank(Key, Rec) ->
 fold_ranks(Fun, Acc, Rec) ->
     gen_fold(ranks, Fun, Acc, Rec).
 
+-spec session_count() -> non_neg_integer().
```

Comment:
Ideally, I would prefer to move this function to mgmt. This module was designed to encapsulate session states completely devoid of the business logic, rather than a convenient place for the helper functions.

If we expose a session count function from this module, to me it means that we commit ourselves to providing session count to business logic. But since the current implementation requires a full sweep of the table, perhaps it's best to avoid it for now.

Reply:
Good point.  I'll fix this in a follow up.